system/Model.php

```diff
@@ -1265,7 +1265,7 @@ protected function doProtectFields(array $data): array
 			throw DataException::forInvalidAllowedFields(get_class($this));
 		}
 
-		if (is_array($data) && count($data))
+		if (is_array($data) && $data)
```

Comment:
Maybe we can dispense with the use of the ` && $data` check? The loop won't execute if we would supply an empty array.

Reply:
updated,  remove unneeed if (is_array($data)), it is always an array as a field parameter